logzio-sender/src/main/java/io/logz/sender/HttpsSyncSender.java

```diff
@@ -118,7 +84,38 @@ public void sendToLogzio(List<FormattedLogMessage> messages) throws LogzioServer
         }
     }
 
-    private void readBadRequest(HttpURLConnection conn) {
+    private int tryBackoff(int currentRetrySleep, int currTry, int responseCode, String responseMessage, IOException savedException) throws LogzioServerErrorException, InterruptedException {
+        if (currTry == configuration.getMaxRetriesAttempts()) {
+            if (savedException != null) {
+                reporter.error("Got IO exception on the last bulk try to logz.io", savedException);
+            }
+            // Giving up, something is broken on Logz.io side, we will try again later
+            throw new LogzioServerErrorException("Got HTTP " + responseCode + " code from logz.io, with message: " + responseMessage);
+        }
+
+        reporter.warning("Could not send log to logz.io, retry (" + currTry + "/" + configuration.getMaxRetriesAttempts() + ")");
+        reporter.warning("Sleeping for " + currentRetrySleep + " ms and will try again.");
+        Thread.sleep(currentRetrySleep);
+        return currentRetrySleep * 2;
+    }
+
+    private boolean shouldRetry(byte[] payload, int responseCode, String responseMessage, HttpURLConnection conn) {
+        boolean retry = false;
+        if (responseCode == HttpURLConnection.HTTP_BAD_REQUEST) {
+            readResponse(conn);
+        }
+        else if (responseCode == HttpURLConnection.HTTP_UNAUTHORIZED) {
+            reporter.error("Logz.io: Got forbidden! Your token is not right. Unfortunately, dropping logs. Message: " + responseMessage);
+        }
+        else if (responseCode == HttpURLConnection.HTTP_OK) {
+            reporter.info("Successfully sent bulk to logz.io, size: " + payload.length);
+        } else {
+            retry = true;
+        }
+        return retry;
+    }
+
+    private void readResponse(HttpURLConnection conn) {
```

Comment:
It's not really reading the response, it's more logging the response as warn. Maybe return reponse and then write it to log ?